src/lib/Cashlink.ts

```diff
@@ -128,6 +143,7 @@ class Cashlink {
         this._getNetwork = () => networkPromise;
 
         if (value) this.value = value;
+        if (fee) this._fee = fee; // set via _fee to avoid throwing an exception in fee setter if already claimed
```

Comment:
Why add a check for immutability and then go around it here? I think this is bad and the immutability check above should just be a warning then.

Reply:
This is to be able to (re)create cashlink instances (for example from the store) that are already claimed and have a fee assigned.
The sanity check is to block changing the fee when it wouldn't have any effect anymore.
The construction however is an atomic operation at which all this is an allowed configuration.

Reply:
I think that actually setting a fee after the cashlink was claimed should not be an error. It can be a warning, but why should it throw?

Reply:
Throwing is more consistent to what the `value` and `message` setters do.
Equally one could discuss whether the `value` and `message` setters should rather warn than throw as changing the value can make sense for cashlinks that should get recharged or be able to be claimed multiple times with smaller amounts. Also changing the cashlink message after creation can still make sense, for example in the gift card app, the cashlink message in the QR code could get updated when the message in the gift card gets changed.
Given these aspects, throwing for `value` and `message` but not for `fee` feels arbitrary to me.
However, in the end, I don't care too much whether it throws or warns.
To get this merged, I changed it.